### PR TITLE
Make TestEventTrigger delegate testable and extensible

### PR DESCRIPTION
**Pull request by tcarmelveilleux**
- Make TestEventTriggerDelegate base class manage multiple sub-handlers.
- Add unit tests for TestEventTriggerDelegate
- Introduce an interface for generic TestEventTrigger handlers
- Add Clear() method to Intrusive list to remove everything.
- Document the fact that `emberAfHandleEventTrigger` should NOT be used
- Update all examples as needed.
  - Renumber OTA and SmokeCO testevent trigger prefix to match new Matter 1.3 standard (first 16 bits == cluster ID being tested).
  - Clean-up dead code
  - Register OTA test event handler as intended.
- Add TODOs for platforms to clean-up TestEventTriggerDelegate registration.
- Updated BOOLCFG tests to use `--hex-arg` for PIXIT since `--int-arg` is too confusing with complex keys.

Issue #31723

Testing done:
- EEVSE Python tests using event triggers pass
- SMOKECO YAML tests using event triggers pass
- BOOLCFG Python tests using event triggers pass
- TC_TestEventTrigger passes
- New unit tests pass

### CODE REVIEW

File: src/app/clusters/smoke-co-alarm-server/SmokeCOTestEventTriggerHandler.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
```
Comment (yunhanw-google) [inline src/app/clusters/smoke-co-alarm-server/SmokeCOTestEventTriggerHandler.h:3]: nit: 2024

**Review by yunhanw-google — APPROVED** [review]

File: src/app/clusters/smoke-co-alarm-server/SmokeCOTestEventTriggerHandler.h (hunk 1 above)
Comment (tcarmelveilleux) [reply src/app/clusters/smoke-co-alarm-server/SmokeCOTestEventTriggerHandler.h:3]: Not a new file, only a rename

**Review by ReneJosefsen — APPROVED** [review]

File: examples/platform/silabs/SilabsTestEventTriggerDelegate.h [hunk 2]
```diff
@@ -18,14 +18,38 @@
...
+/**
+ * @brief User handler for handling the test event trigger
+ *
+ * @note If TestEventTrigger is enabled, it needs to be implemented in the app
+ *
+ * @param eventTrigger Event trigger to handle
+ *
+ * @warning *** DO NOT USE FOR STANDARD CLUSTER EVENT TRIGGERS ***
+ *
+ * TODO(#31723): Rename `emberAfHandleEventTrigger` to `AmebaHandleGlobalTestEventTrigger`
```
Comment (cecille) [inline examples/platform/silabs/SilabsTestEventTriggerDelegate.h:37]:
```suggestion
 * TODO(#31723): Rename `emberAfHandleEventTrigger` to `SilabsHandleGlobalTestEventTrigger`
```

**Review by mkardous-silabs — APPROVED** [review]

File: src/platform/nxp/k32w/k32w1/DefaultTestEventTriggerDelegate.h [hunk 3]
```diff
@@ -25,12 +25,9 @@ namespace chip {
 class DefaultTestEventTriggerDelegate : public TestEventTriggerDelegate
```
Comment (cecille) [inline src/platform/nxp/k32w/k32w1/DefaultTestEventTriggerDelegate.h:25]: Are these platform specific classes doing anything different now than your SimpleTestEventTriggerDelegate?

Also, kQueryTrigger - that seems to have disappeared - was that not used anywhere?

File: examples/platform/silabs/SilabsTestEventTriggerDelegate.h (hunk 2 above)
Comment (tcarmelveilleux) [reply examples/platform/silabs/SilabsTestEventTriggerDelegate.h:37]: Will fix.

File: src/platform/nxp/k32w/k32w1/DefaultTestEventTriggerDelegate.h (hunk 3 above)
Comment (tcarmelveilleux) [reply src/platform/nxp/k32w/k32w1/DefaultTestEventTriggerDelegate.h:25]: - They call `emberAfHandleEventTrigger` which is how those platform examples are hooked-up. The init of the TestEventTriggerDelegate is not in the examples, but rather in the platform main, so I cannot add more handlers and that's how they did the hookups.
- kQueryTrigger was not doing anything, and not in allowable range. Will ask NXP what it did, but certainly nothing in SDK.

File: examples/platform/silabs/SilabsTestEventTriggerDelegate.h (hunk 2 above)
Comment (tcarmelveilleux) [reply examples/platform/silabs/SilabsTestEventTriggerDelegate.h:37]: See https://github.com/project-chip/connectedhomeip/pull/31748

**Pull request merged**